File: leetcode/2610. Convert an Array Into a 2D Array With Conditions/main.dart
class Solution {
  List<List<int>> findMatrix(List<int> nums) {
    List<List<int>> res = [];

    List<int> freq = List<int>.filled(nums.length + 1, 0);

    for (int number in nums) {
        if (freq[number] >= res.length) {
            res.add([]);
        }

        res[freq[number]].add(number);
        freq[number]++;
    }

    return res;
  }
}

File: leetcode/2610. Convert an Array Into a 2D Array With Conditions/main.py
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        res: List[List[int]] = []
        freq = [0] * (len(nums) + 1)

        for num in nums:
            if freq[num] >= len(res):
                res.append([])

            res[freq[num]].append(num)
            freq[num] += 1

        return res
